uncertainties/core.py

```diff
@@ -790,7 +782,7 @@ def std_dev(self, std_dev):
         # (Note: if NaN < 0 is False, there is no need to test
         # separately for NaN. But this is not guaranteed, even if it
         # should work on most platforms.)
-        if std_dev < 0 and not isinfinite(std_dev):
+        if std_dev < 0 and not (isinf(std_dev) or isnan(std_dev)):
```

Comment:
I see that this maintains the current behavior because `math.isinfinite` does not exist. Should we take the opportunity here to do what the code was trying to do like this?

```suggestion
        if std_dev < 0 and isfinite(std_dev):
```

with a `from math import isfinite` above? I think previously `isinfinite` may have been accidentally added instead of `isfinite`.